aws/data_source_aws_route53_delegation_set.go

```diff
@@ -0,0 +1,63 @@
+package aws
+
+import (
+	"fmt"
+	"log"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/service/route53"
+	"github.com/hashicorp/terraform/helper/schema"
+)
+
+func dataSourceAwsDelegationSet() *schema.Resource {
+	return &schema.Resource{
+		Read: dataSourceAwsDelegationSetRead,
+
+		Schema: map[string]*schema.Schema{
+			"id": {
+				Type:     schema.TypeString,
+				Required: true,
+			},
+			"caller_reference": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"name_servers": {
+				Type:     schema.TypeList,
+				Elem:     &schema.Schema{Type: schema.TypeString},
+				Computed: true,
+			},
+		},
+	}
+}
+
+func dataSourceAwsDelegationSetRead(d *schema.ResourceData, meta interface{}) error {
+	conn := meta.(*AWSClient).r53conn
+
+	dSetID := d.Get("id").(string)
+
+	input := &route53.GetReusableDelegationSetInput{
+		Id: aws.String(dSetID),
+	}
+
+	log.Printf("[DEBUG] Reading Route53 delegation set: %s", input)
+
+	req, resp := conn.GetReusableDelegationSetRequest(input)
+	err := req.Send()
+	if err != nil {
+		return fmt.Errorf("Failed getting Route53 delegation set: %s Set: %q", err, dSetID)
+	}
+
+	d.SetId(dSetID)
+	d.Set("caller_reference", resp.DelegationSet.CallerReference)
+
+	servers := []string{}
+	for _, server := range resp.DelegationSet.NameServers {
+		if server != nil {
+			servers = append(servers, "nope")
```

Comment:
Should this be `servers = append(servers, server)`?

Reply:
Oops, yeah I should have threw a WIP tag on this. My bad!  I'm focusing on this PR now and will be making updates to get this ready for final review.

Reply:
This fix is included in my latest commit as well.